content/articles/aes-rsa-encryption/index.md

```diff
@@ -0,0 +1,138 @@
+
+AES and RSA are an example of these algorithms that are used in data encryption in data communication. AES is an acronym that stands for Advanced Encryption Standard while RSA stands for Rivest, Shamir, Adleman.
+
+### Getting started with AES encryption algorithm
+
+It was initially referred to as Rijndael and is a Symmetric [block cipher](https://en.wikipedia.org/wiki/Block_cipher) algorithm. AES was specifically developed for encrypting electronic data back in the United States by the [NIST](https://en.wikipedia.org/wiki/National_Institute_of_Standards_and_Technology). Ever since the AES has been used worldwide in encrypting confidential data.
+
+AES was developed as a replacement for [DES](https://searchsecurity.techtarget.com/definition/Data-Encryption-Standard) after it started showing vulnerabilities to Brute-force attackers.
+
+### Functionality of AES algorithm
+
+The functionality of the AES algorithm might seem complex but in a real sense, it is really simple to understand. The first step is knowing that AES has 3 Block ciphers, namely:
+
+- **AES-128**
+
+As we already know, the encryption and decryption processes are done using a key.
+This block cipher uses a key that is 128 bits long for encryption and decryption of a message.
+- **AES-192**
+
+The AES-192 uses a key that is 192 bits long for the encryption and decryption of messages.
+
+- **AES-256**
+
+The AES-256 uses a key that is 256 bits long for the encryption and decryption of messages.
+
+AES being asymmetric block cipher, means that the key used in encryption and decryption is a kept secret and is only made known to the sender and the recipient of the message. The key used for encrypting the message is usually the same key used to decrypt the message on the receiving end.
+
+When encrypting using AES, a person can either use AES-128, AES-192, or even the AES-256. During this encryption, some steps are followed to encrypt a message and the and these steps are done in rounds. Each round is made up of several steps of Substitution, transposition, and mixing of the message into ciphertext and also has a key called a [round key](https://www.quora.com/What-is-the-round-key-in-an-AES-algorithm).
+
+Below is a simple diagrammatic representation flow of how AES encrypts data:
+
+![AES algorithm flow ](/engineering-education/aes-rsa-encryption/AES_flow.png)
+
+This illustrates that either of the 3 encryption keys is used to encrypt the message. The encryption process occurs in the "cipher" and an encrypted message is produced in the "ciphertext".
+
+The encryption process has 10 rounds for a 128-bit long key, 12 rounds for a 192-bit long key, and 14 rounds for a 256-bit long key.
+
+### Step 1: Generation of round keys
+
```

Comment:
This next steps illustrates......

You need to add a sentence in between the sentence and the header to guide the reader on what's happening in the next step. The transition needs to be smooth.